### PR TITLE
WIP sdk2-final tests

### PR DESCRIPTION
**Pull request by jaekwon**

### CODE REVIEW

File: codec.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 
 type Codec struct {
 	mtx               sync.RWMutex
-	typeInfos         map[reflect.Type]*TypeInfo
+	typeInfos         map[string]*TypeInfo
```
Comment (jaekwon) [inline codec.go:12]: Why?  reflect.Type is comparable.

File: reflect.go [hunk 2]
```diff
@@ -158,10 +158,9 @@ type FieldOptions struct {
 // Usage:
 // `wire.RegisterInterface((*MyInterface1)(nil), nil)`
 func (cdc *Codec) RegisterInterface(ptr interface{}, opts *InterfaceOptions) {
-	cdc.mtx.Lock()
-	defer cdc.mtx.Unlock()
```
Comment (jaekwon) [inline reflect.go:161]: We need these locks.

File: reflect.go [hunk 3]
```diff
@@ -182,8 +184,6 @@ func (cdc *Codec) RegisterInterface(ptr interface{}, opts *InterfaceOptions) {
 // Usage:
 // `wire.RegisterConcrete(MyStruct1{}, "com.tendermint/MyStruct1", nil)`
 func (cdc *Codec) RegisterConcrete(o interface{}, name string, opts *ConcreteOptions) {
-	cdc.mtx.Lock()
-	defer cdc.mtx.Unlock()
```
Comment (jaekwon) [inline reflect.go:187]: ditto

File: reflect.go (hunk 2 above)
Comment (ebuchman) [reply reflect.go:161]: the locks are called in setTypeInfo also which these functions call so this was deadlocking ...

Comment (jaekwon) [reply reflect.go:161]: You're right!

**Comment by jaekwon** [comment]
Closing because replaced by branch sdk2-final-tests

**Pull request closed without merging**